link/perf_event.go

```diff
@@ -204,6 +217,25 @@ func getPMUEventType(pmu string) (uint64, error) {
 	return et, nil
 }
 
+// determineRetprobeBit reads a Performance Monitoring Unit's retprobe (numeric identifier)
+// from /sys/bus/event_source/devices/<pmu>/format/retprobe.
+func determineRetprobeBit(pmu string) (uint64, error) {
+	p := filepath.Join("/sys/bus/event_source/devices/", pmu, "/format/retprobe")
+
+	data, err := os.Open(p)
+	if err != nil {
+		return 0, fmt.Errorf("reading file %s: %w", p, err)
+	}
+
+	var retprobe uint64
+	_, err = fmt.Fscanf(data, "config:%d", &retprobe)
```

Comment:
Newline at the end? I also remember that you have to check n == 1 (return from fscanf) for this to work reliably when the file contents are bogus.